ui/js/keypad.js

```diff
@@ -170,4 +196,24 @@ Keypad.prototype._keyPress = function _keyPress (target) {
   if (this.type === 'usSsn' && isValidSsn((this.result))) {
     this.keypad.find('.enter')[0].disabled = false
   }
+
+  if (this.type === 'custom' && this.opts.constraint === 'date') {
+    this.keypad.find('.enter')[0].disabled = !validateDate(this.result)
+  }
+}
+
+Keypad.prototype.setOpts = function setOpts (newOpts) {
+  this.opts = newOpts
+  if (this.opts.constraint) {
+    switch (this.opts.constraint) {
+      case 'length':
+        LENGTHS.custom = this.opts.maxLength
+        break
+      case 'date':
+        LENGTHS.custom = 8
+        break
+      default:
+        break
+    }
+  }
```

Comment:
Wouldn't it be better to just ditch this function when you find out `this.opts.constraint` is not set?

```suggestion
Keypad.prototype.setOpts = function setOpts (newOpts) {
  this.opts = newOpts
  if (!this.opts.constraint) return
  
  switch (this.opts.constraint) {
      case 'length':
        LENGTHS.custom = this.opts.maxLength
        break
      case 'date':
        LENGTHS.custom = 8
        break
      default:
        break
    }
```